### ISSUE TITLE
Standard actions as cards

### ISSUE THREAD

**Issue by sienmich** [bafolts/terraforming-mars#2192]
Once #2185 is done, I'm thinking about making standard actions also as cards.

I mean those 4:
1. Convert plants to greenery
2. Convert heat to temp
3. Trade with colony
4. Send a delegate

And those 2 (they would require some nice graphic imo, sb have an idea / can help?)
5. Claim a milestone
6. Fund an award

They all work just like standard project, except the fact they are not. So I think having them in front and back end as some StandardAction which is parent class to the new StandardProject would be nice. Any thoughts? :smile:

**Comment by sienmich** [comment]
And possibly we can do sth with kelvinist / scientists action too?

**Comment by bafolts** [comment]
The code separation will be great for this as it will take logic out of `Player` and `Game` again. Displaying the standard actions as cards on the UI makes sense. On the back end I would start to separate them from cards or subclass card in a way where parts can be shared. The `IActionCard` really has nothing to do with a card. I would consider renaming this `IAction` and begin moving away from everything being named `Card` when they are not cards. Similar to the `CardManifest`. That should really be `ExpansionManifest`.

**Comment by kevinb9n** [comment]
+1
Completely agree, the SP vs SA distinction is arbitrary and what's good for one is good for the other!

Does it gray out the ones I can't currently afford?

If you are willing to show three different cards for Trade, then there won't need to be a second step to choose what to pay.

I assume that Send Delegate will just have no cost before the arrow if I have someone in the lobby.

> And those 2 (they would require some nice graphic imo, sb have an idea / can help?)
> 5. Claim a milestone
> 6. Fund an award

You can just do what the Vitor card is doing.

**Comment by kevinb9n** [comment]
(Whoever plays this as Ecoline first will complain that it doesn't change to 7 plants for them (and Cryo Sleep etc.) but I wouldn't suggest doing that unless we think we're gonna do cost changes more comprehensively like in #2187 / #776.)

**Comment by kevinb9n** [comment]
How about I file a separate issue exploring how we deal with special party-based actions (of which there are now many, if PolitAgen becomes official)?

**Comment by kevinb9n** [comment]
It sounds like our top-level "your turn" menu is reducing down to just:

* Play card
* Use card action
* Standard action
* Standard project
* (and the non-action options: end, pass, undo)

(I'm assuming for now we can fold the turmoil actions in somewhere appropriate, and I'm also not worrying about the special first action of corps like Inventrix.)

1. It might be better to have just one option "standard action" open up *both* lists at once, as long as the lists are separate and with headings. That way, new players don't have to remember which kind of thing is which kind of thing.

2. This makes me wonder if it could work to just add two more "cards" to the standard actions, one saying "(red-arrow) Play Card" and one saying "Use Card Action", and then this whole thing essentially *becomes* the top-level menu (and replaces #953).  The non-action options (end/pass/undo) need to appear there too but I don't think they should look at all like cards.

3. This all makes for a big list, and it suggests we might want to switch how we show the text, to the way milestones, awards, and soon ruling parties (#1696) do it.
